gno.land/pkg/integration/testing_integration.go

```diff
@@ -211,10 +230,73 @@ func setupGnolandTestScript(t *testing.T, txtarDir string) testscript.Params {
 
 				tsValidateError(ts, "gnokey", neg, err)
 			},
+			"sleep": func(ts *testscript.TestScript, neg bool, args []string) {
```

Comment:
i don't remember how, but I think sleep was already possible.
iirc, we inherit txtar builtin functions too.

cc @gfanton

Reply:
No, it's not built-in, but thinking more about it, are you sure this is mandatory? I'm kind of scared to add this feature. I've removed it for a good reason after a discussion with @thehowl. It depends on the use case, but I think it can be a bad habit to have this, and it would be better to base tests on events rather than waiting and expecting things to happen. Also, if people abuse this feature, we might end up having very slow tests.

Reply:
I can give an example.  Imagine there is a realm with a function that can be called.  There is a time duration interval defined.  Maybe this function accepts an integer and sums all values it is provided with. But every 10 seconds the function resets the summed value to zero.  The reset is triggered by a call to this function.  It would look something like this:
```go
package sumtimer

import "time"

var (
    sum int
    startTime
)

func AddToSum(i int) {
    if time.Since(startTime) > time.Second * 10 {
        sum = 0
        startTime = time.Now
    }

    sum += i
}
```
It would be impossible to run automated tests on code like this without `sleep`.  We can get rid of it if there is a strong opposition to it but I think it can be very useful.  Like any other code that is committed, hopefully the reviewers would spot when someone is abusing this.  I'd actually prefer that there are no tests committed using this code; it is just nice to have it so that, for example, I can write automated tests that run locally.

Reply:
Hmm, I'm still not convinced. 10 seconds is already pretty huge, and what if we have a delay of 1 hour or even 2 days for whatever reason? We're not going to use a `sleep` command to test that, right? Wouldn't it be better to unit test this method by using a sub-unexported method with a timeout as an argument? I really think that `txtar` should not be used for time-based tests. event if it works for some edge cases, I think this will introduce bad habits generally. wdyt ?

Reply:
What if we keep the code but comment it out so people can just uncomment it to use it for testing locally?  In my opinion it's more convenient testing using txtar than it is to deploy locally and verify results manually.

Reply:
The easiest way to test it is by writing a unittest and using the std.TestSkip* helpers. These helpers simulate a sleep from the contract's point of view, allowing testing for extended periods of time (hours). However, it makes sense to have a single integration test of ~3-5 seconds in txtar if it's the only one.

Reply:
Oh awesome; I didn't know that existed. I think using that inside a unit test addresses my motivation for this feature so I'll remove `sleep`

Reply:
You don't need a custom txtar command to add a pause, you can just add this line to your txtar:

```
exec sleep 10
```

The `exec` command allows you to run any existing shell command.